packages/mui-base/src/TextareaAutosize/TextareaAutosize.tsx

```diff
@@ -122,71 +118,26 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(
 
     // Take the box sizing into account for applying this value as a style.
     const outerHeightStyle = outerHeight + (boxSizing === 'border-box' ? padding + border : 0);
-    const overflow = Math.abs(outerHeight - innerHeight) <= 1;
+    const overflowing = Math.abs(outerHeight - innerHeight) <= 1;
 
-    return { outerHeightStyle, overflow };
+    return { outerHeightStyle, overflowing };
   }, [maxRows, minRows, props.placeholder]);
 
-  const updateState = (prevState: State, newState: State) => {
-    const { outerHeightStyle, overflow } = newState;
-    // Need a large enough difference to update the height.
-    // This prevents infinite rendering loop.
-    if (
-      renders.current < 20 &&
-      ((outerHeightStyle > 0 &&
-        Math.abs((prevState.outerHeightStyle || 0) - outerHeightStyle) > 1) ||
-        prevState.overflow !== overflow)
-    ) {
-      renders.current += 1;
-      return {
-        overflow,
-        outerHeightStyle,
-      };
-    }
-    if (process.env.NODE_ENV !== 'production') {
-      if (renders.current === 20) {
```

Comment:
We still need a failsafe for unstable layouts.

Reply:
> for unstable layouts (unrelated to React). This happens when the CSS values are unstable,

Can you provide reproductions of layout instability not related to React? I'm open to debugging it.

> this warning is also meant to invite developers to create issues so we can debug and fix the logic

Sure, but since we're not relying on React state anymore, there's no need for warnings about excessive re-renders caused by state.

> Typically you would see the height flicker up and down as you type in the input. Or freeze the app because the resize observer logic run in a loop.

Again, can you provide reproductions using this PR build?